web/pdf_viewer.js

```diff
@@ -309,7 +309,17 @@ class PDFViewer {
       this.renderingQueue = options.renderingQueue;
     }
 
-    this.scroll = watchScroll(this.container, this._scrollUpdate.bind(this));
+    const { abortController } = options;
+    abortController?.signal.addEventListener("abort", () => {
+      this.#resizeObserver.disconnect();
+      this.#resizeObserver = null;
+    });
```

Comment:
Should this listener perhaps be removed once invoked?
```suggestion
    }, { once: true });
```